api/internal/git/cloner.go

```diff
@@ -29,43 +29,57 @@ func ClonerUsingGitExec(repoSpec *RepoSpec) error {
 
 	cmd := exec.Command(
 		gitProgram,
-		"clone",
-		"--depth=1",
-		repoSpec.CloneSpec(),
+		"init",
 		repoSpec.Dir.String())
 	out, err := cmd.CombinedOutput()
 	if err != nil {
-		log.Printf("Error cloning git repo: %s", out)
+		log.Printf("Error initializing git repo: %s", out)
 		return errors.Wrapf(
 			err,
-			"trouble cloning git repo %v in %s",
-			repoSpec.CloneSpec(), repoSpec.Dir.String())
+			"trouble initializing git repo in %s",
+			repoSpec.Dir.String())
 	}
 
+	cmd = exec.Command(
+		gitProgram,
+		"remote",
+		"add",
+		"origin",
+		repoSpec.CloneSpec())
+	cmd.Dir = repoSpec.Dir.String()
+	out, err = cmd.CombinedOutput()
+	if err != nil {
+		log.Printf("Error adding remote: %s", out)
+		return errors.Wrapf(err, "trouble adding remote %s", repoSpec.CloneSpec())
+	}
+
+	ref := "HEAD"
```

Comment:
Maybe use `origin/HEAD`?

Reply:
`ref` is used with `git fetch origin <ref>` so `origin` is already specified. Using `origin/HEAD` actually fails to fetch anything.